### DISCUSSION TITLE
Ender 3 S1 with the STM32F401RCT6 SoC

### DISCUSSION THREAD

**Discussion by TopazAaron** [mriscoc/Ender3V2S1#106] [General]
Just curious if there is any hope for us with the STM32F4 and if there is I can test the firmware, was just curious on a time line maybe.

**Comment by mriscoc** [comment]
I could compile a version for the STM32F4 but maybe that board have a different pin assigment.

**Reply by TopazAaron** [reply]
I'm willing to try, I have a file for the stock firmware if it would help? I've had to flash it a few times, I tired flashing your firmware before I realized I had the f4 chip lol

**Reply by mriscoc** [reply]
We must have access to the pin assignment of that board or to the oficial sources for that version.

**Reply by bankm1** [reply]
I'm in the same situation with the f4 chip. How can I determine the pinout of this board?

**Reply by mriscoc** [reply]
Not the pin out, the SoC pin assignment, i.e. what pins are used for motor drivers, end-stops, etc.

**Reply by stanleygreenwood** [reply]
Hi Miguel, 

Is there any way for someone with a board to get this information for you OR is it only Creality that can provide this?
I am fairly technical and can assist if it is a possibility.

Thank you in advance!

**Reply by mriscoc** [reply]
Hi, We have to know the contents of the file: pins_CREALITY_S1_F401RC.h or verify the SoC pin routing with an ohmmeter (multimeter), also there are some issues with the programming environment used for compile for this board.

**Comment by mriscoc** [comment]
EDIT2: New version released here: https://github.com/mriscoc/Ender3V2S1/discussions/106#discussioncomment-2597577
EDIT: Rename the firmware file to **firmware.bin**

This is a private build for testing the preliminary compatibility of the firmware with the **Ender3S1** with the **F4 SoC**, please use it with caution.

### Disclaimer
THIS FIRMWARE AND ALL OTHER FILES IN THE DOWNLOAD ARE PROVIDED FREE OF CHARGE WITH NO WARRANTY OR GUARANTEE. SUPPORT IS NOT INCLUDED JUST BECAUSE YOU DOWNLOADED THE FIRMWARE. WE ARE NOT LIABLE FOR ANY DAMAGE TO YOUR PRINTER, PERSON, OR ANY OTHER PROPERTY DUE TO THE USE OF THIS FIRMWARE. IF YOU DO NOT AGREE TO THESE TERMS THEN DO NOT USE THIS FIRMWARE.

**Reply by houseofbugs** [reply]
Can you post source for this bin file that you've distributed? Much appreciated.

**Reply by mriscoc** [reply]
Yes, after the tests I will post the results here: https://github.com/mriscoc/Ender3V2S1 you can for now use the sources from https://github.com/CrealityOfficial/Ender-3S1 I got a confirmation that sources compile and flash correctly for the F4 SoC.

**Reply by houseofbugs** [reply]
I have the F4 and it's not flashing. Hence why I was asking for the source that it was built with. My board also will not take from the Creality repo either.

**Reply by Geo-Warrior** [reply]
Hi. I'm aslo using the F4 board and upon trying to upgrade using the test firmware the printer boots up like usual and is still showing the v3.0.3 as current firmware. Is there anything I can do to help?

**Reply by mriscoc** [reply]
I will try to do more tests later.

**Reply by theSt33v** [reply]
It worked for me. I renamed the file to firmware.bin before flashing, although I'm not sure how much that matters.

**Reply by mriscoc** [reply]
That's very important finding, please post a capture of your info page.

**Reply by azsundevil02** [reply]
I have one S1 with the board that accepted the firmware and one s1 that no matter what I do - it will not take any firmware - even creality's.   I've tried 8 different cards and formatting and file names etc etc - still nothing.

I don't think it's the firmware - think it's something with the boards possibly.  

On this firmware - o did have to power cycle a couple times before it read the card and updated on the one that worked.

**Reply by theSt33v** [reply]
Can you think of anything that may have happened with the good one that didn't happen with the bad one? For example, did you ever brick and have to recover the good one due to a board firmware mismatch?

**Reply by azsundevil02** [reply]
The "good one" was able to update to Creality firmware 3.1.0 previously - the "bad one" would not update firmware no matter what I did.  

I did brick both of them and the both could be restored with the 3.0.3 firmware version that had the laser attachment.

I opened them both up today and swapped motherboard as the "bad one" is being returned to Amazon as it kept having z offset issues and driving itself into the bed and changing z offsets every couple
Prints.

Creality did not assemble the internals very well as a fuse on the bad one was not in all the way and was crooked.  They both had a lot of hot glue around all the connections.

I checked and both boards and chips look the same and have the same markings / labels 

I thought if o adjusted the connections and fixed the crooked fuse that maybe that would get the bad one to take firmware - but no.  Even after booting up and trying 20 times - it still will not take any firmware from Creality or the professional. I've tried adjusting and renaming the files, reformatting, different cards and the original cards.

I even tried the card / firmware that worked on the good one and then t didn't work.

I'm at a loss why mine and some others will not accept firmware updates.

**Reply by azsundevil02** [reply]
Oh and I updated the display on the good one and tried on the bad one - the good one took it and the bad one didn't do anything - so not sure if it took it.

Did not do anything with the boot loader that was included with the update folder.

**Reply by houseofbugs** [reply]
I think we either have a case of Creality messing up the bootloader on these or they are trying to lock us out. I think I am just going to focus my efforts on an OpenBLT based bootloader so we can just reprogram it and be done with it.

**Reply by houseofbugs** [reply]
For the record. I have 20 new boards with the F4 CPUs here that I got directly from Creality last week. None will take *any* firmware.

**Reply by azsundevil02** [reply]
It wouldn't shock me at all if Creality were doing that - they are pushing their cloud service and their wifi box really hard.....

**Reply by theSt33v** [reply]
It was mentioned earlier that a board that won't otherwise take firmware can still be restored from a soft brick. That means the bootloader isn't totally non-functional. Maybe you could brick the board and then install whatever compatible firmware you want instead of restoring the one that's on there.

Also, the fact that it was soft bricked in the first place means that flashing firmware via the SD card is possible on that board, or it was originally at least.

**Reply by azsundevil02** [reply]
I tried that on the one that wouldn't work. It didn't work for me (may work for others) - but I had to use a specific card with the previous firmware it had had loaded or the laser version from Creality.  

Others may have different results....

**Comment by theSt33v** [comment]
Here you are.

Also, it's important to note that you sent me this build on Facebook. I didn't download it from GitHub. The two filenames have the same date so I'm sure they're the same, but now you know in case they're not.

Steps I did to flash:

1. Rename file to firmware.bin
2. Placed onto empty SD card.
3. Put card in powered-off printer.
4. Powered on.

![PXL_20220418_011755902](https://user-images.githubusercontent.com/21251502/163743597-82fbfc02-40f2-4fac-9a50-5276e81375a4.jpg)

**Reply by mriscoc** [reply]
It is the same file, thanks for your confirmation.

**Comment by mriscoc** [comment]
New Experimental version with the changes from the APRIL2022 release:  

[Experimental-Ender3S1-STM32F4-20220420.zip](https://github.com/mriscoc/Ender3V2S1/files/8523879/Experimental-Ender3S1-STM32F4-20220420.zip)

@houseofbugs has discover that the firmware must be inside the STM32F4_UPDATE folder of the memory card for a correct flash procedure in some S1 F4 printers.

This is a private build for testing the preliminary compatibility of the firmware with the **Ender3S1** with the **F4 SoC**, please use it with caution.

### Disclaimer
THIS FIRMWARE AND ALL OTHER FILES IN THE DOWNLOAD ARE PROVIDED FREE OF CHARGE WITH NO WARRANTY OR GUARANTEE. SUPPORT IS NOT INCLUDED JUST BECAUSE YOU DOWNLOADED THE FIRMWARE. WE ARE NOT LIABLE FOR ANY DAMAGE TO YOUR PRINTER, PERSON, OR ANY OTHER PROPERTY DUE TO THE USE OF THIS FIRMWARE. IF YOU DO NOT AGREE TO THESE TERMS THEN DO NOT USE THIS FIRMWARE.

**Reply by mnelson7982** [reply]
I can confirm a successful installation on my machine.  I dropped the .bin in the root of the SD and it worked as expected.

**Reply by azsundevil02** [reply]
I can install the 04/19 version - but cannot install the version with UBL in the name on my F4 board S1.

**Reply by mriscoc** [reply]
> I can confirm a successful installation on my machine. I dropped the .bin in the root of the SD and it worked as expected.

So, you didn't need to use the STM32F4_UPDATE folder? can you post a capture of your info page?

**Reply by azsundevil02** [reply]
Yes sir!!

But no matter what I do the UBL version will not load - I soft brick" the board.

I have been having some issues with mesh and the machine not following the m420 on prints - so wanted to try the UBL version.

Still loving the firmware!

![image](https://user-images.githubusercontent.com/79348716/164290682-35999f46-2560-4dbb-a774-0726cb511bcb.jpeg)

**Reply by mriscoc** [reply]
> Yes sir!!
> 
> But no matter what I do the UBL version will not load - I soft brick" the board.
> 
There isn't a S1F4 UBL version, the only UBL version is for the S1F1. Also, there is a problem with the F4 and the amount of flash memory for the firmware, currently the UBL version does not fit within the available area.

**Reply by mnelson7982** [reply]
> > I can confirm a successful installation on my machine. I dropped the .bin in the root of the SD and it worked as expected.
> 
> So, you didn't need to use the STM32F4_UPDATE folder? can you post a capture of your info page?

that’s correct. I added the .bin to the root of the SD card. 

![image](https://user-images.githubusercontent.com/1141858/164291633-9f1331c7-2437-4658-893d-d369f2633942.jpg)

**Reply by iamjawnwick** [reply]
This worked for me! My S1 doesn't require the bin to be in a folder.

**Comment by fabiomaticus** [comment]
I have successfully installed the firmware on my S1 F4. I found a problem with bed tramming: it does not register, and the message "Corner leveling ..." appears.

**Reply by mriscoc** [reply]
Please post a capture of your info page

**Reply by azsundevil02** [reply]
I was able to update the firmware directly from the root of the card as well.  If I put it in the folder my printer ignored it all together and booted up in the previously installed firmware.

![image](https://user-images.githubusercontent.com/79348716/164290543-b44e7117-5130-433f-b641-906cd16d0254.jpeg)

**Reply by fabiomaticus** [reply]
> Please post a capture of your info page

This is my
![WhatsApp Image 2022-04-20 at 20 55 49](https://user-images.githubusercontent.com/32363332/164303095-c745baec-99c5-47b3-a2c1-365c1476ca5c.jpeg)
.

**Comment by azsundevil02** [comment]
Ahhh -

Yes - user error!

Ok - makes sense.

Right now I’m just trying to figure out why my prints won’t use the m420 on prints - corners and z offset look good and then center is not low enough and may prints fail.

I digress - sorry - that wasn’t what this thread was about.

If I put I’m the firmware in a folder it ignores it.

If I put it on the root it took it - but it took a couple tries and I had to rename it to firmware42.bin.

All the others just say on the boot screen.

This newer one does seem more snappy / responsive on my machine by the way.

Rick

**Comment by mnelson7982** [comment]
<img width="1023" alt="image" src="https://user-images.githubusercontent.com/1141858/164308579-c7ab9fd5-f186-4981-bb49-6dd80e649e27.png">

This makes me so incredibly happy. Running the Ender3S1 without Octoprint felt like the printer was incomplete.  Having Octoprint working without limitations is such a relief. 

Thank you @mriscoc for all the effort here 🥇
